### PR TITLE
Feature/local metrics output

### PR DESCRIPTION
**Pull request by alonbentamar**
Save the metrics and metadata to the local profile files

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have updated the relevant documentation.
- [ ] I have added tests for new logic.

### CODE REVIEW

File: gprofiler/merge.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 from gprofiler.gprofiler_types import ProcessToStackSampleCounters, StackToSampleCount
 from gprofiler.log import get_logger_adapter
 from gprofiler.metadata.metadata_type import Metadata
+from gprofiler.system_metrics import Metrics
```
Comment (Jongy) [inline gprofiler/merge.py:17]: Oof, I don't like it that we have to change this file at all. The "metadata" and "metrics" is unrelated, logically, to this file.

@alonbentamar think if you have any way to change the API such that adding new metadata/metrics/whatever doesn't involve touching this file.

**Review by Jongy — CHANGES_REQUESTED** [review]
1 small, if you can't find a good way to do that, let's merge

File: gprofiler/merge.py (hunk 1 above)
Comment (alonbentamar) [reply gprofiler/merge.py:17]: Meh, this file is the one that outputs the final stack collapse, I don't think it's bad that it knows about the metadata and metrics as they are part of the file

Comment (Jongy) [reply gprofiler/merge.py:17]: It's not "bad" but it's not useful either, these are different parts of the code. `merge.py` is about the merging of stackcollapse files, not appending random data as the header in files. Can we maybe extract that logic away from this file? It's something that I'd wanted to do a while back.

Comment (alonbentamar) [reply gprofiler/merge.py:17]: It's pretty messy right now, can't think of a nice way to do it without a pretty big refactor :/

Comment (Jongy) [reply gprofiler/merge.py:17]: We'll need to do that one soon

**Review by Jongy — APPROVED** [review]

**Pull request merged**